osdk/src/config/manifest.rs

```diff
@@ -0,0 +1,356 @@
+                scheme.inherit(&current_manifest.default_scheme);
+            }
+            current_manifest
+        }
+
+        // Search for OSDK.toml in the current directory first.
+        let current_manifest_path = PathBuf::from("OSDK.toml").canonicalize().ok();
+        let mut current_manifest = match &current_manifest_path {
+            Some(path) => deserialize_toml_manifest(path),
+            None => None,
+        };
+        // Then search in the workspace root.
+        let workspace_manifest_path = workspace_root.join("OSDK.toml").canonicalize().ok();
+        // The case that the current directory is also the workspace root.
+        if let Some(current) = &current_manifest_path {
+            if let Some(workspace) = &workspace_manifest_path {
+                if current == workspace {
+                    return finalize(current_manifest);
+                }
+            }
+        }
+        let workspace_manifest = match workspace_manifest_path {
+            Some(path) => deserialize_toml_manifest(path),
+            None => None,
+        };
+        // The current manifest should inherit settings from the workspace manifest.
+        if let Some(workspace_manifest) = workspace_manifest {
+            if current_manifest.is_none() {
+                current_manifest = Some(workspace_manifest);
+            } else {
+                // Inherit one scheme at a time.
+                let current_manifest = current_manifest.as_mut().unwrap();
+                current_manifest
+                    .default_scheme
+                    .inherit(&workspace_manifest.default_scheme);
+                for (scheme_string, scheme) in workspace_manifest.map {
+                    let current_scheme = current_manifest
+                        .map
+                        .entry(scheme_string)
+                        .or_insert_with(Scheme::empty);
+                    current_scheme.inherit(&scheme);
+                }
+            }
+        }
+        finalize(current_manifest)
+    }
+
+    /// Get the scheme given the scheme from the command line arguments.
+    pub fn get_scheme(&self, scheme: Option<impl Display>) -> &Scheme {
+        if let Some(scheme) = scheme {
+            let selected_scheme = self.map.get(&scheme.to_string());
+            if selected_scheme.is_none() {
+                error_msg!("Scheme `{}` not found in `OSDK.toml`", scheme);
+                process::exit(Errno::ParseMetadata as _);
+            }
+            selected_scheme.unwrap()
+        } else {
+            &self.default_scheme
+        }
+    }
+}
+
+fn deserialize_toml_manifest(path: impl AsRef<Path>) -> Option<TomlManifest> {
+    if !path.as_ref().exists() || !path.as_ref().is_file() {
+        return None;
+    }
+    // Read the file content
+    let contents = fs::read_to_string(&path).unwrap_or_else(|err| {
+        error_msg!(
+            "Cannot read file {}, {}",
+            path.as_ref().to_string_lossy().to_string(),
```

Comment:
`Cow<str>` also implements `Display` trait. So it's useless to call `to_string` after `to_string_lossy`.